core-bundle/templates/Error/service_unavailable.html.twig

```diff
@@ -4,5 +4,36 @@
     {{ 'XPT.unavailable'|trans }}
 {% endblock %}
 {% block matter %}
-    <p>{{ 'XPT.maintenance'|trans }}</p>
+    <p data-trans="XPT.maintenance">{{ 'XPT.maintenance'|trans }}</p>
+{% endblock %}
+
+{% block body_end %}
+    <script id="maintenance-script" type="text/javascript" data-labels="{{ defaultLabels|json_encode() }}">
+        (function (n, d) {
+            let language = n.language;
+            let data = JSON.parse(d.getElementById('maintenance-script').dataset.labels);
+            function translate(key) {
+                if (data[language] === undefined) {
+                    language = new Intl.Locale(language).language;
+                    if (data[language] === undefined) {
+                        return null;
+                    }
+                }
+                if (data[language][key] === undefined) {
+                    return null;
+                }
+                return data[language][key];
+            }
+            d.querySelectorAll('[data-trans]').forEach(el => {
+                let trans = translate(el.dataset.trans);
+                if (trans) {
+                    el.innerHTML = trans;
```

Comment:
```suggestion
                    el.textContent = trans;
```

If I’m right, the messages get encoded in the twig template, so we should not allow HTML here I think.